### DISCUSSION TITLE
external js not working on build

### DISCUSSION THREAD

**Discussion by bvsicvllywhvtever** [gatsbyjs/gatsby#34327] [Help]
Hello!

I'm a young developer who picked up this site from another team who left me with no documentation, and I'm trying to make edits to the site with very little knowledge of GatsbyJS, so please bear with me.

I've been trying to make a dropdown trigger on the click of an a tag. I created a new static folder at the root of my directory, placed the new js file in there, and linked it in a react helmet on the page like so:

`	    <Helmet>
		<script src={withPrefix('/collapse.js')} type="text/javascript" />
	    </Helmet>`

Here are the contents of the collapse.js file if that helps:
```
var coll = document.getElementsByClassName("team-member__link");
var i;

for (i = 0; i < coll.length; i++) {
  coll[i].addEventListener("click", function() {
    this.classList.toggle("active");
    var content = this.nextElementSibling;
    var extender = document.getElementsByClassName("extend")[0];
    if(extender.style.display === "block"){
	extender.style.display = "none";
    }
    if (content.style.display === "block") {
      content.style.display = "none";
    } else {
      content.style.display = "block";
      for(j = 0; j < coll.length; j++){
	if(coll[j] == this && j > 3){
		document.getElementsByClassName("extend")[0].style.display = "block";
	}
	if(coll[j] != this){
		coll[j].nextElementSibling.style.display = "none";
	}
      }
    }
  });
}
```

Everything works fine locally, but the js strangely won't work when I push to my staging site and the server adds a trailing slash to the url (/example vs /example/). The trailing slash doesn't affect anything locally.

I've tried to see if there is a way to turn off the automatic trailing slash on the server, but the platform doesn't allow for it, and I worry there might be some security risk with trying to force a redirect in the code (again this is stuff totally out of my wheelhouse, so I apologize if that is a ridiculous concern). I've seen some forums mentioning that JS only loads once with GatsbyJS, so I'm wondering if that might be an issue. If so, I have no idea how to go about it.

If anyone has any idea what's going on, please let me know. I'm totally stumped on this one.

**Comment by AnilSeervi** [comment]
try it without the forward-slash at the beginning
```jsx
import { withPrefix } from "gatsby"
// ...

<Helmet> 
    <script src={withPrefix('collapse.js')} type="text/javascript" />
</Helmet>
```

**Reply by bvsicvllywhvtever** [reply]
Still nothing.

**Reply by AnilSeervi** [reply]
https://github.com/gatsbyjs/gatsby/discussions/27889#discussioncomment-254854
This might help :)

**Reply by bvsicvllywhvtever** [reply]
Thank you for the link. It helped clear a little bit up about the trailing slashes.

However, I'm not sure the trailing slashes are being added in Gatsby, but instead are being added by the hosting provider. There's not a way to turn this off with the provider (it's some obscure provider called [render](https://render.com/)). Do you think some kind of internal RegEx redirect would fix this or would it create an infinite loop (and if it doesn't how would I go about adding this in my project)? Or is there a different way to include the external js that would work once deployed?
